document-store/src/main/java/org/hypertrace/core/documentstore/postgres/query/v1/vistors/PostgresRelationalIdentifierAccessingExpressionVisitor.java

```diff
@@ -0,0 +1,37 @@
+package org.hypertrace.core.documentstore.postgres.query.v1.vistors;
+
+import static org.hypertrace.core.documentstore.postgres.utils.PostgresUtils.prepareCast;
+import static org.hypertrace.core.documentstore.postgres.utils.PostgresUtils.prepareFieldDataAccessorExpr;
+
+import org.hypertrace.core.documentstore.expression.impl.IdentifierExpression;
+import org.hypertrace.core.documentstore.expression.type.SelectTypeExpression;
+import org.hypertrace.core.documentstore.postgres.query.v1.PostgresQueryParser;
+
+public class PostgresRelationalIdentifierAccessingExpressionVisitor
+    extends PostgresSelectTypeExpressionVisitor {
+  private final String baseField;
+  private final SelectTypeExpression rhs;
+
+  public PostgresRelationalIdentifierAccessingExpressionVisitor(
+      final PostgresSelectTypeExpressionVisitor baseVisitor,
+      final String baseField,
+      final SelectTypeExpression rhs) {
+    super(baseVisitor);
+    this.baseField = baseField;
+    this.rhs = rhs;
+  }
+
+  @Override
+  public PostgresQueryParser getPostgresQueryParser() {
+    return baseVisitor.getPostgresQueryParser();
```

Comment:
Same question as above: https://github.com/hypertrace/document-store/pull/191#discussion_r1452928412

Reply:
Actually, that is also not required as we are not passing in any baseVisitor in the constructor. Updated there to get rid of the unnecessary conditional statement.